gradle/publish.gradle

```diff
@@ -66,10 +68,35 @@ task prepareGhCredentials {
 prepareGhPages.dependsOn groovydoc
 publishGhPages.dependsOn prepareGhCredentials
 
+task prepareGradlePluginsRepoRelease {
+    description = "Prepare the release to the Gradle Plugins Repository"
+    group = 'release'
+    doLast {
+        def secretsProperties = rootProject.file('secrets.properties')
+        if (secretsProperties.exists()) {
```

Comment:
I believe we can just assume that we *always* expect the credentials to be part of `secrets.properties` and let the `gradle-build-properties-plugin` to deal with that. This task's action would become:
```
// force evaluation of properties to check whether they're there or not
buildProperties.secrets['gradle.publish.key'].string
buildProperties.secrets['gradle.publish.secret'].string
System.properties['com.gradle.login.properties.file'] = rootProject.file('secrets.properties')
```